codesign/bin/codesign.dart

```diff
@@ -2,8 +2,99 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'package:codesign/codesign.dart' as codesign;
+import 'dart:io';
 
-void main(List<String> arguments) {
-  print('Hello world: ${codesign.calculate()}!');
+import 'package:args/args.dart';
+import 'package:codesign/codesign.dart';
+import 'package:platform/platform.dart';
+
+/// Definitions of variables are included in help texts below.
+const String helpFlag = 'help';
+const String kCommit = 'commit';
+const String kProduction = 'production';
+const String kCodesignCertName = 'codesign-cert-name';
+const String kCodesignUserName = 'codesign-username';
+const String kAppSpecificPassword = 'app-specific-password';
+const String kCodesignAppStoreId = 'codesign-appstore-id';
+const String kCodesignTeamId = 'codesign-team-id';
+
+/// Perform Mac code signing based on file paths.
+///
+/// If `--production` is set to true, code signed artifacts will be uploaded
+/// back to google cloud storage.
+/// Otherwise, nothing will be uploaded back for production. default value is
+/// false.
+///
+/// For `--commit`, asks for the engine commit to be code signed.
+///
+/// For `--filepath`, provides the artifacts zip paths to be code signed.
+///
+/// Usage:
+/// dart run bin/main.dart --commit=a5967ed309ef2beb9625f128571f7060597b5eda
```

Comment:
Avoid putting real data in here. It will make it a bit too easy to accidentally do the wrong thing.

```suggestion
/// ```
///   dart run bin/main.dart --commit=$commitSha \
///   --filepath=darwin-x64/FlutterMacOS.framework.zip \
///   --filepath=ios/artifacts.zip \
///   --filepath=dart-sdk-darwin-arm64.zip
/// ```
```